pkg/kubelet/stats/cadvisor_stats_provider_test.go

```diff
@@ -92,7 +93,16 @@ func TestFilterTerminatedContainerInfoAndAssembleByPodCgroupKey(t *testing.T) {
 			t.Errorf("%q is expected to be in the output\n", c)
 		}
 	}
-	for _, c := range []string{"pod0-i-terminated-1", "pod0-c0-terminated-1", "pod0-i-terminated-2", "pod0-c0-terminated-2", "pod0-i", "pod0-c0", "c1"} {
+
+	expectedInfoKeys := []string{"pod0-i-terminated-1", "pod0-c0-terminated-1", "pod0-i-terminated-2", "pod0-c0-terminated-2", "pod0-i", "pod0-c0"}
+	// NOTE: on Windows, IsSystemdStyleName will return false, which means that the Container Info will
+	// not be assembled by cgroup key.
+	if goruntime.GOOS != "windows" {
+		expectedInfoKeys = append(expectedInfoKeys, "c1")
+	} else {
+		expectedInfoKeys = append(expectedInfoKeys, "pod1-c1.slice")
+	}
```

Comment:
should we call the "IsSystemdStyleName" function to decide which name to add, rather than an OS check?